### PR TITLE
Add dependabot auto-merge

### PR DESCRIPTION
**Pull request by austinvazquez**
*Issue #, if available:*
N/A

*Description of changes:*
Add auto-merge of dependabot pull-requests.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/automerge.yml
Comment (swagatbora90) [inline .github/workflows/automerge.yml:25]: Should we protect the `amazon-ecr-credential-helper` main branch first, before enabling this workflow? We should make sure that the status checks pass before a patch gets merged.

**Review by swagatbora90 — CHANGES_REQUESTED** [review]

File: .github/workflows/automerge.yml
Comment (austinvazquez) [reply .github/workflows/automerge.yml:25]: Yeah sure, my initial impression from my research was this only merge after status checks, but let me verify that behavior. I might just need to add a sync step between the workflows. Something like if all checks pass, approve and merge.

Comment (austinvazquez) [reply .github/workflows/automerge.yml:25]: Oh I understand now after a second read of https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions#enable-auto-merge-on-a-pull-request

I'll look at our settings.

Comment (austinvazquez) [reply .github/workflows/automerge.yml:25]: @swagatbora90, I updated branch protections to require status check pass before merging and require one approval.

Comment (swagatbora90) [reply .github/workflows/automerge.yml:25]: Also, should we limit this to only minor or patch version upgrades?  Major version upgrade can introduce  breaking changes, so it may be better to have some manual review. The above link does something similar.

```
- name: Enable auto-merge for Dependabot PRs
        if: steps.metadata.outputs.update-type == 'version-update:semver-patch'}}
        run: gh pr merge --auto --merge "$PR_URL"
        env:
          PR_URL: ${{github.event.pull_request.html_url}}
          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
```

**Review by swagatbora90 — COMMENTED** [review]
LGTM.

File: .github/workflows/automerge.yml
Comment (austinvazquez) [reply .github/workflows/automerge.yml:25]: That makes sense to me. Let's start with just minor and patch and see if we can reduce some of the manual review. Off the top of my head, AWS SDK pushes lots of minor and patch updates, but like you mention a major update would probably be best for us to take a look.

**Review by swagatbora90 — APPROVED** [review]
LGTM!

**Review by ginglis13 — APPROVED** [review]

**Pull request merged**
